### PR TITLE
feat: featureflag for auto tagging T0 AD group members

### PR DESCRIPTION
**Pull request by JonasBK**
## Description

Add a feature flag to enable/disable auto-tagging of Tier Zero AD group members

Ticket: BP-702

## Motivation and Context

Some users (mostly BHE users) prefer no automatic Tier Zero tagging of Tier Zero AD group members as they would like to manually verify new members of Tier Zero groups. The feature flag allows them to disable the auto-tagging.

For BHE users, disabling the member auto-tagging will result in MemberOf findings, which comes with the BHE pull request 536. That PR should be merged at the same time as this PR.

## How Has This Been Tested?

Tested locally. Disabled and enabled it a couple of times to confirm it works.

We should perform a performance test with BHE to confirm that analysis does not increase significantly with this change. 

## Types of changes

-   [ ] Chore (a change that does not modify the application functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [ ] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: cmd/api/src/daemons/datapipe/agi.go [hunk 1]
```diff
@@ -160,11 +160,13 @@ func TagActiveDirectoryTierZero(ctx context.Context, db database.Database, graph
 
 	if autoTagT0ParentObjectsFlag, err := db.GetFlagByKey(ctx, appcfg.FeatureAutoTagT0ParentObjects); err != nil {
 		return err
+	} else if autoTagMembersFlag, err := db.GetFlagByKey(ctx, appcfg.FeatureAutoTagT0ADMembers); err != nil {
```
Comment (juggernot325) [inline cmd/api/src/daemons/datapipe/agi.go:163]: I tried to test this locally and was unable to build the app. There's a change in main that breaks this line. The `db` variable is no longer present in the latest code. It looks like it was changed in #816.

**Review by juggernot325 — CHANGES_REQUESTED** [review]

File: cmd/api/src/daemons/datapipe/agi.go (hunk 1 above)
Comment (JonasBK) [reply cmd/api/src/daemons/datapipe/agi.go]: Fixed

**Comment by JonasBK** [comment]
Issue identified. Will investigate.

**Comment by JonasBK** [comment]
Closing this PR as we will implement the feature later in another way.

**Pull request closed without merging**